src/app/qgsmaptoolmovefeature.cpp

```diff
@@ -187,12 +187,20 @@ void QgsMapToolMoveFeature::cadCanvasReleaseEvent( QgsMapMouseEvent *e )
 
     vlayer->beginEditCommand( mMode == Move ? tr( "Feature moved" ) : tr( "Feature copied and moved" ) );
 
+    QgsFeatureRequest request;
+    request.setFilterFids( mMovedFeatures );
     switch ( mMode )
     {
       case Move:
-        for ( QgsFeatureId id : std::as_const( mMovedFeatures ) )
+      {
+        QgsFeatureIterator fi = vlayer->getFeatures( request );
```

Comment:
We can squeeze a bit more performance here by skipping the fetching of attributes, since we don't require them to change the feature geometry:

```suggestion
        QgsFeatureRequest request;
        request.setFilterFids( mMovedFeatures ).setNoAttributes();
        QgsFeatureIterator fi = vlayer->getFeatures( request );
```

Reply:
Done, I forgot about it. Thanks